src/astro/sql/operators/sql_decorator.py

```diff
@@ -81,27 +213,41 @@ def hook(self):
         )
 
     def execute(self, context: Dict):
+
+        # check for string/sqla based sql queries
         if not isinstance(self.sql, str):
             cursor = self._run_sql(self.sql, self.parameters)
             if self.handler is not None:
                 return self.handler(cursor)
             return cursor
 
+        # init db related vars
         self._set_variables_from_first_table()
         self.database = self.database or self.database_from_conn_id
 
+        # init db var - schema, user
         conn = BaseHook.get_connection(self.conn_id)
-
         self.schema = self.schema or SCHEMA
         self.user = conn.login
 
-        self.convert_op_arg_dataframes()
-        self.convert_op_kwarg_dataframes()
-        self.read_sql()
+        # place all the dataframe into sql table from op_kwargs/op_args
+        self.load_dataframes_from_op_args_to_sql_table()
+        self.load_dataframes_from_op_kwargs_to_sql_table()
+
+        # get sql query
+        self.get_sql_query_and_parameters()
+
+        # handle templating
         self.handle_params(context)
+
+        # update table name in context based on database
         context = self._add_templates_to_context(context)
+
+        # render templating in sql query
         if context:
             self.sql = self.render_template(self.sql, context)
+
+        # update parameters for snowflake table with fully qualified names
         self._process_params()
 
         output_table_name: str = ""
```

Comment:
It seems like a decent amount of logic can be shared between the sql_decorator and sql_executor. Might be worth looking into how we can abstract more?